eth/tracers/native/prestate.go

```diff
@@ -102,12 +106,18 @@ func (t *prestateTracer) CaptureStart(env *vm.EVM, from common.Address, to commo
 	toBal := new(big.Int).Sub(t.pre[to].Balance, value)
 	t.pre[to].Balance = toBal
 
-	// The sender balance is after reducing: value and gasLimit.
+	// The sender/payer balance is after reducing: value and gasLimit.
 	// We need to re-add them to get the pre-tx balance.
 	fromBal := new(big.Int).Set(t.pre[from].Balance)
 	gasPrice := env.TxContext.GasPrice
 	consumedGas := new(big.Int).Mul(gasPrice, new(big.Int).SetUint64(t.gasLimit))
-	fromBal.Add(fromBal, new(big.Int).Add(value, consumedGas))
+	diffFromBal := new(big.Int).Set(value)
+	if t.payer != nil {
+		t.pre[*t.payer].Balance = new(big.Int).Add(t.pre[*t.payer].Balance, consumedGas)
+	} else {
```

Comment:
Could we a little comment to explain the logic here? It's better to refactor latter